SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1447 @@
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass
+	)
+
+	$passwordPolicyNodes = New-Object System.Collections.Generic.List[System.Object]
+	$count = 0
+
+    Try {
+		$checkServer = Test-Connection -ComputerName $server -Quiet -Count 1
+		if ($checkServer -eq "True") {
+			while ($checkServer -le 2) {
+				Write-LogMessage -Type INFO -Message "Attempting to connect to NSX Edge '$server'"
+				$NSXToken = Request-NsxToken -fqdn $server -username $user -password $pass 
+				if ($NSXToken) {
+					$nsxtEdgeNodes = (Get-NsxtEdgeCluster | Where-Object {$_.member_node_type -eq "EDGE_NODE"})
+					foreach ($nsxtEdgeNode in $nsxtEdgeNodes.members) {
+						$nsxtConfigTypeValues =  New-Object -TypeName psobject
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "transportNodeID" -notepropertyvalue "Empty"
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "passwdMinimumLength" -notepropertyvalue "Empty"
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "cliPasswdMaxFailAttempts" -notepropertyvalue "Empty"
+						$nsxtConfigTypeValues | Add-Member -notepropertyname "cliPasswdMaxFailIntervalInSec" -notepropertyvalue "Empty"
+
+						$response = Get-NsxtEdgeNodeAuthPolicy -nsxtManager $server -nsxtEdgeNodeID $nsxtEdgeNode.transport_node_id
+						$nsxtConfigTypeValues.transportNodeID = $nsxtEdgeNode.transport_node_id
+						$nsxtConfigTypeValues.passwdMinimumLength = $response.minimum_password_length
+						$nsxtConfigTypeValues.cliPasswdMaxFailAttempts = $response.cli_max_auth_failures
+						$nsxtConfigTypeValues.cliPasswdMaxFailIntervalInSec = $response.cli_failed_auth_lockout_period  
+
+						$nsxtEdgeNodeAdds = Get-NsxtEdgeNode -transportNodeID $nsxtEdgeNode.transport_node_id
+						$nodeConfigValue = New-Object -TypeName psobject
+						$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $nsxtEdgeNodeAdds.node_deployment_info.node_settings.hostname
+						$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "NSXEdge"
+						$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $nsxtConfigTypeValues
+						$passwordPolicyNodes.Add($nodeConfigValue)
+						Remove-Variable -Name nodeConfigValue
+						Remove-Variable -Name nsxtConfigTypeValues
+					}
+					return $passwordPolicyNodes
+				} else {
+					if ($ErrorLog -match '"error_code":404') {
+						$count += 1
+						Write-LogMessage -Type WARNING -Message "Unable to submit request. Retry again in 20 secs (Try $count/3)"
+						Start-Sleep -s 20
+					} else {
+						Write-LogMessage -Type INFO -Error "Failed to obtain access token from NSX Edge '$server',  please verify your credentials and try again"  -ErrorAction Stop
+					}
+				}
+			}
+			Write-LogMessage -Type Error -Message "Failed to obtain access token from NSX Edge '$server',  please verify your credentials and try again" -ErrorAction Stop
```

Comment:
Would color be included?

Reply:
addressed in bbfbaf6dac6bb885aa39e716c9d8971c0a5a58d7